### PR TITLE
Writing field related to company-dependent one doesn't work 

### PR DESCRIPTION
**Pull request by yelizariev**
This PR adds a test that shows the problem

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by yelizariev** [comment]
The issue is potentially related to #35659 

PS @beledouxdenis you can understand know why I [dived](https://github.com/odoo/odoo/pull/35659#discussion_r400896328) in test_new_api module:-)

**Comment by yelizariev** [comment]
The logs from runbot:
```

Build #1973704 Creating test subuild with config "All at_install" (at_install tests only) 
odoo.addons.test_new_api.tests.test_new_fields:1046 test_27_company_dependent
FAIL: TestFields.test_27_company_dependent
Traceback (most recent call last):
  File "/data/build/odoo/odoo/addons/test_new_api/tests/test_new_fields.py", line 1046, in test_27_company_dependent
    self.assertEqual(attribute_record.company_foo, 'XYZ')
AssertionError: 'DEF' != 'XYZ'
- DEF
+ XYZ
```

File: odoo/addons/test_new_api/models.py [hunk 1]
```diff
@@ -452,6 +452,7 @@ class CompanyDependentAttribute(models.Model):
     _description = 'Test New API Company Attribute'
 
     company = fields.Many2one('test_new_api.company')
+    company_foo = fields.Char(related='company.foo')
```
Comment (Feyensv) [inline odoo/addons/test_new_api/models.py:455]: The field has to be specified as `readonly=False` ;). (Just tested locally and the test works fine after this change)

Comment (yelizariev) [reply odoo/addons/test_new_api/models.py:455]: Oh, it's just new default readonly=True value for related fields. Thank you!

**Comment by pedrobaeza** [comment]
@yelizariev FYI that was introduced in v12. We mentioned it on the OCA migration steps: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0#tasks-to-do-in-the-migration

**Pull request closed without merging**